### PR TITLE
[witness] Move file i/o to main function for easier testing

### PR DESCRIPTION
**Pull request by smeiklej**

### CODE REVIEW

File: witness/golang/cmd/witness/impl/witness.go [hunk 1]
```diff
@@ -55,23 +53,15 @@ type ServerOpts struct {
...
-	fileData, err := ioutil.ReadFile(configFile)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read from config file: %v", err)
-	}
-	var js LogConfig
-	if err := json.Unmarshal(fileData, &js); err != nil {
-		return nil, fmt.Errorf("failed to parse config file as proper JSON: %v", err)
-	}
+// loadConfig loads the log configuration information into a map.
+func loadConfig(config LogConfig) (map[string]witness.LogInfo, error) {
```
Comment (AlCutter) [inline witness/golang/cmd/witness/impl/witness.go:61]: Maybe we could rename this to be something like `buildLogMap` or similar?

File: witness/golang/cmd/witness/main.go [hunk 2]
```diff
@@ -44,12 +46,24 @@ func main() {
 		glog.Exitf("Error forming a signer: %v", err)
 	}
 
+	if len(*configFile) == 0 {
+		glog.Exitf("--config_file must not be empty")
```
Comment (AlCutter) [inline witness/golang/cmd/witness/main.go:50]: supernit: `glog.Exit()` (no format params)

**Review by AlCutter — COMMENTED** [review]
Just a couple of tiny suggestions

File: witness/golang/cmd/witness/impl/witness.go (hunk 1 above)
Comment (smeiklej) [reply witness/golang/cmd/witness/impl/witness.go]: Sure, done.

File: witness/golang/cmd/witness/main.go (hunk 2 above)
Comment (smeiklej) [reply witness/golang/cmd/witness/main.go]: I was just copying the change you made a few lines above :).  But cool, changed them both now.

**Review by AlCutter — APPROVED** [review]

**Pull request merged**